sdk/src/utils/is-response.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Custom type guard to check if an object is likely a Fetch Response
+ */
+export function isFetchResponse(result: unknown): boolean {
```

Comment:
```suggestion
export function isFetchResponse(result: unknown): result is Response {
```

With this, we could omit https://github.com/directus/directus/pull/19898/files#diff-3d8c3738e2feee86a090991f222b35ff4f8e7b6f3d110e0f33d7aa15b82701f7R10